### ISSUE TITLE
More differences in instance and subclass checks between typing.Union and types.Union

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#88834]
BPO | [44668](https://bugs.python.org/issue44668)
--- | :---
Nosy | @gvanrossum, @serhiy-storchaka, @Fidget-Spinner

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-07-18.17:21:30.520>
labels = ['type-bug', 'library', '3.10', '3.11']
title = 'More differences in instance and subclass checks between typing.Union and types.Union'
updated_at = <Date 2021-07-19.17:57:02.931>
user = 'https://github.com/serhiy-storchaka'
```

bugs.python.org fields:
```python
activity = <Date 2021-07-19.17:57:02.931>
actor = 'gvanrossum'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2021-07-18.17:21:30.520>
creator = 'serhiy.storchaka'
dependencies = []
files = []
hgrepos = []
issue_num = 44668
keywords = []
message_count = 4.0
messages = ['397757', '397775', '397778', '397823']
nosy_count = 3.0
nosy_names = ['gvanrossum', 'serhiy.storchaka', 'kj']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue44668'
versions = ['Python 3.10', 'Python 3.11']
```

</p></details>

### Linked PRs
* gh-128363
* gh-128370
* gh-128371

**Comment by AlexWaygood** [comment]
This looks to have been fixed at some point, though I don't know when. Here's the behaviour on 3.11.2:

```pycon
>>> from typing import TypeVar
>>> T = TypeVar("T")
>>> issubclass(int, str | T | int)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 1654, in __subclasscheck__
    if issubclass(cls, arg):
       ^^^^^^^^^^^^^^^^^^^^
TypeError: issubclass() arg 2 must be a class, a tuple of classes, or a union
>>> isinstance(1, str | T | int)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 1650, in __instancecheck__
    return self.__subclasscheck__(type(obj))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 1654, in __subclasscheck__
    if issubclass(cls, arg):
       ^^^^^^^^^^^^^^^^^^^^
TypeError: issubclass() arg 2 must be a class, a tuple of classes, or a union
```

Depending on whether it was fixed "on purpose" or "by accident", we may need to add regression tests.

**Comment by serhiy-storchaka** [comment]
Indeed, the first part was fixed in #91603.

#128363 fixes the second part.
